modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java

```diff
@@ -882,9 +903,14 @@ private ObjectRelationship _addObjectRelationship(
 			long userId, long objectDefinitionId1, long objectDefinitionId2,
 			long parameterObjectFieldId, String deletionType,
 			Map<Locale, String> labelMap, String name, boolean reverse,
-			String type)
+			boolean system, String type)
 		throws PortalException {
 
+		if (system && !SystemUtil.allowManageSystemEntities()) {
```

Comment:
Question to Product Team: Is adding a system object relationship from any object definition possible? Or is it going to be only allowed to modifiable system object definition?

Reply:
Answer: Only Modifiable System Object Definition can create system object relationships